doc/hardware/porting/board_porting.rst

```diff
@@ -956,3 +956,77 @@ extension.
    Board extensions are not allowed in any module referenced in Zephyr's
    ``west.yml`` manifest file. Any board changes are required to be submitted to
    the main Zephyr repository.
+
+.. _board_terminology:
+
+Terminology
```

Comment:
sorry if I missed something but this list is almost exactly a copy-paste of the "Boards, SoCs, etc." section, so I am not sure there is a reason for the two to still co-exist?

Reply:
there was a discussion on having a terminology section, see https://discord.com/channels/720317445772017664/1194671118422913114/1214257604205744158 and before comments. A proposal to which I agree, especially as it also makes it easier to reference.

But you're right that `Boards, SoCs, etc.` contains part of same, but the reason I didn't remove it from there is because I still think it makes sense to have a brief description at that location where we introduces the concepts.

At the same time, a reader will expect that a terminology section describes all important terms, as a form of quick reference. So hard to argue those term should not be there.

Reply:
I think we can live with this duplication. Longer term what we need is a linkable global Glossary instead, like we do in NCS:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/glossary.html

Reply:
> I think we can live with this duplication. Longer term what we need is a linkable global Glossary instead, like we do in NCS:
> https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/glossary.html
> 

Exactly :) What's wrong with using the glossary we already have? The text that's used in "Boards, SoCs, ..." can stay there as a legend for the diagram, but should also link to proper entries in the glossary (and this will be a good opportunity to update existing entries such as "SoC")
https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/glossary.rst